src/qx/Instructions.cpp

```diff
@@ -41,7 +41,7 @@ void Measure::execute(SimulationIterationContext &context) {
 
 
 Reset::Reset(std::optional<core::QubitIndex> qubitIndex)
-    : qubitIndex{ std::move(qubitIndex) }
```

Comment:
Just for my understanding: What is the function of move and why should it be removed here?

Reply:
When you move an object A to an object B, you are basically reusing the space of A in the computer's memory, you are just telling B to point to where A is (then A becomes invalid, of course). Move semantics were introduced in C++11 as an optimization, let's say. Before, either you copied A into B, or you passed a reference from A to  B.

Linters sometimes advise to change from this:

```
B::B(A const& a)
    : a_{ a }
```

To this:

```
B::B(A a)
    : a_{ std::move(a) }
```

In this case, it was telling me that the `move` was unnecessary because the object A was trivially  copyable. Imagine it like if it were an `int` or a `float`. You don't really need to move those. And `QubitIndex` is basically a `size_t`, an `unsigned int`.